src/Expression/StructuredExpression.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Expression;
+
+use Traversable;
+use Yiisoft\Db\Schema\Column\ColumnSchemaInterface;
+
+use function array_key_exists;
+use function array_keys;
+use function is_iterable;
+use function iterator_to_array;
+
+/**
+ * Represents a structured type SQL expression.
+ *
+ * @see https://en.wikipedia.org/wiki/Structured_type
+ *
+ * For example:
+ *
+ * ```php
+ * new StructuredExpression(['price' => 10, 'currency_code' => 'USD']);
+ * ```
+ *
+ * Will be encoded to `ROW(10, USD)` in PostgreSQL.
+ */
+final class StructuredExpression implements ExpressionInterface
+{
+    /**
+     * @param ColumnSchemaInterface[] $columns
+     * @psalm-param array<string, ColumnSchemaInterface> $columns
+     */
+    public function __construct(
```

Comment:
Need phpdoc here badly. It's not obvious about what to pass where. Duplicating docs from getters is alright for this case.

Reply:
Done